cmd/node-termination-handler.go

```diff
@@ -200,7 +200,16 @@ func main() {
 			// Exit interruption loop if a SIGTERM is received or the channel is closed
 			break
 		default:
-			drainOrCordonIfNecessary(interruptionEventStore, *node, nthConfig, nodeMetadata, metrics)
+			for event, ok := interruptionEventStore.GetActiveEvent(); ok && !event.InProgress; event, ok = interruptionEventStore.GetActiveEvent() {
+				select {
+				case interruptionEventStore.Workers <- 1:
+					event.InProgress = true
+					go drainOrCordonIfNecessary(interruptionEventStore, event, *node, nthConfig, nodeMetadata, metrics)
```

Comment:
I think we should also add a WaitGroup to this now that we spawn go-routines for `drainOrCordonIfNecessary`. It's a pretty long function so if a SIGTERM comes in we may shut those go routines down before they're able to complete.  

There's a lot of params to `drainOrCordonIfNecessary` right now and a WaitGroup would add yet another one... But I think we can deal with cleaning that up in a later PR, and maybe make it a little nicer too by doing a single function interface that's pluggable on the drain and carries some of the state or helpers in a struct. 

I would propose a WaitGroup outside of the ticker for-loop and then a `Wait()` after the log, on line 215. What do you think?

Reply:
Thank you @bwagner5 for review and suggestions.

I tried to wrap my head around that idea of using WaitGroup, but in all respect I cannot see how it fits here, maybe you can explain more?
Waitgroups are designed to provide a semaphore in order to track the go-routines to be finished. However, this is the opposite of this concept here, where we do not want to wait on them to be finished, rather the opposite allowing us to continue and offload the main loop from the inidividual event handling.

Maybe you are heading more for a `sync.Pool` of workers idea, where we have a set of available async processors?
That would be possible. but also here I am not seeing how it would help much, but if you prefer I can implement that